filament/src/RenderPass.h

```diff
@@ -209,12 +209,11 @@ class RenderPass {
     }
 
     struct PrimitiveInfo { // 24 bytes
-        FMaterialInstance const* mi = nullptr;                          // 8 bytes (4)
-        backend::Handle<backend::HwRenderPrimitive> primitiveHandle;    // 4 bytes
-        backend::RasterState rasterState;                               // 4 bytes
-        uint16_t index = 0;                                             // 2 bytes
-        Variant materialVariant;                                        // 1 byte
-        uint8_t reserved[13 - sizeof(void*)] = {};                      // 5 byte (9)
+        FRenderPrimitive const* primitive = nullptr;    // 8 bytes (4)
```

Comment:
Unfortunately the reserved space is not enough to store `FRenderPrimitive const* primitive`. We need 8 bytes but 4 bytes are remained. So if we keep `FMaterialInstance const* mi` and add `FRenderPrimitive const* primitive` then the size of `PrimitiveInfo` should be increased. @romainguy said that it's a big increase. Is it fine the size of `PrimitiveInfo` is increased?

Reply:
Ah yes. I really don't like this extra dereference in this main loop. Let me think about this more.

Reply:
Okay I think what we should do at this point in increase the size of `PrimitiveInfo` (like in your first attempt) to 40 bytes, i.e. add the `FRenderPrimitive` pointer + 8 bytes of extra padding.
This will increase the size of `Command` to 48 bytes (from 32), which is I think is acceptable. We might want to increase the size of the command buffer in `Allocator.h`, I would set `FILAMENT_PER_RENDER_PASS_ARENA_SIZE_IN_MB` to 3 and `FILAMENT_PER_FRAME_COMMANDS_SIZE_IN_MB` to 2 -- that's just an increase of 1 MB, it's not the end of the world.

Reply:
Sounds good! It's done.